### ISSUE TITLE
[RLlib] TorchMultiCategorical: input_lens must sum to logits.shape[1]

### ISSUE THREAD

**Issue by JulianLorenz** [ray-project/ray#39234]
### What happened + What you expected to happen

I try to train an agent using PPO in an environment with a MultiDiscrete action space, and I stumbled upon the following error:

TorchMultiCategorical in file `rllib/models/torch/torch_distributions.py` expects tensors to have 2 dimensions of size _env_steps x sum(input_lens)_. In the learner classes, however, its methods get called with tensors with 3 dimensions. This causes to trigger the assertion in `TorchMultiCategorical.from_logits(...)`:
`input_lens must sum to logits.shape[1]`

The corresponding code for tensorflow looks very similar, so I would expect similar errors.

Suggested fix: 
Option 1:
Change TorchMultiCategorical to expect the action dimension in the last dimension (dim=-1) instead of the second dimension

Option 2: 
Call functions of TorchMultiCategorical only with tensors with two dimensions.

### Versions / Dependencies

ray: 2.6.3
Python: 3.10

### Reproduction script

```python
import ray
from ray.rllib.algorithms.ppo import PPOConfig
from ray.rllib.examples.env.random_env import RandomEnv
from gymnasium.spaces import MultiDiscrete

ray.init()

action_space = MultiDiscrete([6,7])

algo = (
    PPOConfig()
    .training(model={"fcnet_hiddens": [256, 256], "use_lstm": True})
    .environment(env=RandomEnv, env_config={"action_space": action_space, "max_episode_len":100, "p_terminated":0.0})
    .build()
)

algo.train()
```

### Issue Severity

High: It blocks me from completing my task.

**Comment by JulianLorenz** [comment]
When creating the reproduction script, I noticed that the behaviour only occurs with `"use_lstm": True` in the model config.

**Comment by sven1977** [comment]
PR in review. Thanks @JulianLorenz for filing this issue :)

**Comment by ArturNiederfahrenhorst** [comment]
Thanks for filing the issue :)

**Comment by sven1977** [comment]
PR has been merged into master. Closing this issue.
